hypershift-operator/controllers/scheduler/dedicated_request_serving_nodes.go

```diff
@@ -473,7 +547,14 @@ func (r *DedicatedServingComponentSchedulerAndSizer) Reconcile(ctx context.Conte
 				return ctrl.Result{}, fmt.Errorf("failed to get nodes from placeholders: %w", err)
 			}
 			if len(candidateNodes) > 0 {
+				pairLabel = candidateNodes[0].Labels[OSDFleetManagerPairedNodesLabel]
+				if pairLabel == "" {
+					return ctrl.Result{}, fmt.Errorf("node %s has no pair label", candidateNodes[0].Name)
+				}
 				log.WithValues("pairLabel", candidateNodes[0].Labels[OSDFleetManagerPairedNodesLabel]).Info("claiming candidate nodes")
+				if err := r.ensurePairConfigMap(ctx, pairLabel, hc.Namespace, hc.Name); err != nil {
```

Comment:
I would expect this controller only ensures the configMap now (not ensureHostedClusterLabelAndTaint). Then we'd run a separate controller to watch those configMaps (e.g. caching by a particular label) and let it ensureHostedClusterLabelAndTaint triggered by the cm events.

Reply:
Hmm, yup I guess that would simplify things. As it is, we're doing too much. Let me rework this a bit.